### ISSUE TITLE
Fix code points sequence rendering (emoji modifiers) (faulty Regex)

### ISSUE THREAD

**Issue by iBicha** [iBicha/EmojiTexture#4]
Combinations are currently not working.
E.g. an emoji of man + skin tone is not detected as a single emoji, but as 2.

**Comment by iBicha** [comment]
Caused by the Regex function when `SanitizeText` is set to true

**Comment by sandiz** [comment]
hi, thanks for the awesome plugin! works great with standard emojis!

do you know what regex string to use to detect emoji + skin tone ? any idea if the regex will be on a per emoji basis or can we catch any emoji + skin tone pair ?

**Comment by iBicha** [comment]
@sandiz thank you for the comment.

Thing is, I started working on this plugin before realizing how complex emojis can be.
For example, there are "woman + skin tone" 👩🏽, and there are "woman + mechanic + skin tone" 👩🏽‍🔧or it could be a freaking entire family 👨‍👨‍👧‍👧which is really each an emoji, with a special character that acts as a separator. Even flags are composed. From what I've been, there is no consistent way that I can use to detect a full sequence that represents one glyph or image. And even if I figure it out, it would be a b**** to maintain since it's constantly evolving. Nonetheless, I'm addressing the issue.

So far, this is the current direction:
If the `SanitizeText ` is set to `true`, the text will be rendered as is.
If not :

- Scan for the first emoji unicode char
- Read `16` characters starting from the found emoji (because I don't think any emoji could be larger than 16 characters)
- Mesure the rendered text. If the text is too wide (aspect ratio width/height) remove one character and mesure again.
- If the aspect ratio is close to a square, it means that probably that sequence is a single emoji glyph, and that's what needs to be rendered.
- The search for the next emoji will be right after that sequence.

I made few tests, and it seems to work well. It has a small overhead, since we are measuring multiple times the size, but it's a fair compromise for now, until I figure out another solution (and that's what the `SanitizeText` is for, to turn this off if you want)

The other problem is that this will not work in TextMesh Pro. It keeps a map from One Unicode character to a sprite, and it replaces it when it is found. Probably will not work, until TextMesh Pro gets updated to support this.

A solution is in the work, but I'm open to suggestions.
